contracts/ATM/ATMGovernance.sol

```diff
@@ -0,0 +1,281 @@
+pragma solidity 0.5.17;
+
+// External Libraries
+import "@openzeppelin/contracts/utils/Address.sol";
+
+// Common
+import "../util/AddressArrayLib.sol";
+import "../util/AddressLib.sol";
+
+// Contracts
+import "@openzeppelin/contracts/access/roles/SignerRole.sol";
+
+// Interfaces
+import "./IATMGovernance.sol";
+
+
+/**
+    @notice This contract is used to modify Risk Settings, CRA or DataProviders for a specific ATM.
+    @author develop@teller.finance
+ */
+contract ATMGovernance is SignerRole, IATMGovernance {
+    // TODO: Inject Settings
+
+    using AddressArrayLib for address[];
+    using AddressLib for address;
+    using Address for address;
+
+    /* Constants */
+
+    /* State Variables */
+
+    // List of general ATM settings. We don't accept settings equal to zero.
+    // Example: supplyToDebtRatio  => 5044 = percentage 50.44
+    // Example: supplyToDebtRatio => 1 = percentage 00.01
+    mapping(bytes32 => uint256) public generalSettings;
+
+    // List of Market specific Asset settings on this ATM
+    // Asset address => Asset setting name => Asset setting value
+    // Example 1: USDC address => Risk Premium => 2500 (25%)
+    // Example 2: DAI address => Risk Premium => 3500 (35%)
+    mapping(address => mapping(bytes32 => uint256)) public assetMarketSettings;
+
+    // List of ATM Data providers per data type
+    mapping(uint8 => address[]) public dataProviders;
+
+    // Unique CRA - Credit Risk Algorithm github hash to use in this ATM
+    string public cra;
+
+    /* External Functions */
+
+    /**
+        @notice Adds a new General Setting to this ATM.
+        @param settingName name of the setting to be added.
+        @param settingValue value of the setting to be added.
+     */
+    function addGeneralSetting(bytes32 settingName, uint256 settingValue)
+        external
+        onlySigner()
+    {
+        require(settingValue > 0, "GENERAL_SETTING_MUST_BE_POSITIVE");
+        require(settingName != "", "GENERAL_SETTING_MUST_BE_PROVIDED");
```

Comment:
Looks like this is repeated a bit, maybe a modifier?

Reply:
I really like this improvement, I created a task to implement this suggestion: https://teller-finance.atlassian.net/browse/PROTOCOL-172